File: 1252-cells-with-odd-values-in-a-matrix/1252-cells-with-odd-values-in-a-matrix.py
class Solution:
    def oddCells(self, m: int, n: int, indices: List[List[int]]) -> int:
        matrix = [ [0] * n for _ in range(m) ]

        for rc in indices:
            for i in range(n):
                matrix[rc[0]][i] += 1
            for i in range(m):
                matrix[i][rc[1]] += 1
        
        return sum(cell % 2 != 0 for row in matrix for cell in row)
        

File: 1252-cells-with-odd-values-in-a-matrix/1252-cells-with-odd-values-in-a-matrix.rb
# @param {Integer} m
# @param {Integer} n
# @param {Integer[][]} indices
# @return {Integer}
def odd_cells(m, n, indices) #m=col|n=row
  matrix = Array.new(m){Array.new(n){0}}
  indices.each {|rc|
    (0...n).each do |i| 
      matrix[rc[0]][i] += 1
    end
    (0...m).each do |i|
      matrix[i][rc[1]] += 1
    end
  }
  matrix.flatten.count{|c| c.odd?}
end